### PR TITLE
WIP refactor project layout 

### PR DESCRIPTION
**Pull request by chrisnojima**
This is a WIP.

Theory:
Move desktop into shared
Move react-native into shared
Remove shared symlinks from desktop/react-native
Fixup links in android/ios projects

Pros:
1. No symlinks for electron
1. Structure similar to 'standard' projects

Cons:
1. Shared package.json for electron and react-native

TODO:
- [x] Get android studio working with packager
- [x] Retest electron
- [x] Retest ios
- [ ] Android prod build
- [ ] ios prod build
- [x] Electron prod build
- [ ] Split commits: moves, path changes, etc

### CODE REVIEW

File: shared/desktop/webpack.config.base.js [hunk 1]
```diff
@@ -34,26 +34,26 @@ module.exports = {
     libraryTarget: 'commonjs2',
   },
   resolve: {
-    modulesDirectories: [path.join(__dirname, 'node_modules')],
-    root: [path.join(__dirname)],
+    // modulesDirectories: [path.join(__dirname, 'node_modules')],
```
Comment (chromakode) [inline shared/desktop/webpack.config.base.js]: What's the status of this commented-out code? To be removed?

File: shared/package.json [hunk 2]
```diff
@@ -38,17 +37,24 @@
     "vendor-install": "node npm-vendor.js",
     "vendor-update": "node npm-vendor.js update",
     "watch-test-file": "KEYBASE_RUN_MODE=devel npm run _helper watch-test-file",
-    "wrap": "npm uninstall fsevents && npm prune && npm dedupe && npm shrinkwrap --dev"
+    "wrap": "npm uninstall fsevents && npm prune && npm dedupe && npm shrinkwrap --dev",
+    "rn-start": "./node_modules/react-native/packager/packager.sh",
+    "rn-gobuild-ios": "./react-native/gobuild.sh ios",
+    "rn-gobuild-android": "./react-native/gobuild.sh android",
+    "rn-packager-wipe-cache": "node ./react-native/wipe-cache.js",
+    "rn-test-ios": "xcodebuild test -project ios/Keybase.xcodeproj -scheme 'Keybase For Test' -destination 'platform=iOS Simulator,name=iPhone 6s,OS=9.3'"
```
Comment (chromakode) [inline shared/package.json]: `ios/Keybase.xcodeproj` -> `./react-native/ios/Keybase.xcodeproj`?

File: shared/engine/index.js [hunk 3]
```diff
@@ -306,6 +306,7 @@ const makeEngine = () => {
 
   if (!engine) {
     engine = (process.env.KEYBASE_NO_ENGINE || isTesting) ? new FakeEngine() : new Engine()
+    console.log('aaa', engine)
```
Comment (chromakode) [inline shared/engine/index.js]: ?

File: shared/desktop/webpack.config.base.js (hunk 1 above)
Comment (chrisnojima) [inline shared/desktop/webpack.config.base.js]: yeah, i'll do a cleanup pass

File: shared/package.json (hunk 2 above)
Comment (chrisnojima) [inline shared/package.json]: i'll fix. the testing stuff is pretty broken so i didn't bother. when fixing up the project i killed the test target even

File: shared/engine/index.js (hunk 3 above)
Comment (chrisnojima) [inline shared/engine/index.js]: i'll remove this was just me trying to debug why i was getting duplicate engines

Comment (chrisnojima) [inline shared/engine/index.js]: done

**Comment by chrisnojima** [comment]
I'm going to wait until yarn to do this

**Comment by chrisnojima** [comment]
- [ ] Old jenkins builds back

**Pull request closed without merging**